magentoerpconnect_pricelist_currency/model/sale.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2015 FactorLibre (http://www.factorlibre.com)
+#                  Hugo Santos <hugo.santos@factorlibre.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields
+from openerp.addons.connector.unit.mapper import mapping
+from openerp.addons.magentoerpconnect.backend import magento
+from openerp.addons.magentoerpconnect.sale import SaleOrderImportMapper, \
+    SaleOrderImporter
+
+
+class MagentoSaleOrder(models.Model):
+    _inherit = 'magento.sale.order'
+
+    magento_currency_id = fields.Many2one('res.currency', 'Magento Currency')
+
+
+@magento(replacing=SaleOrderImportMapper)
```

Comment:
When you use `@magento(replacing=SaleOrderImportMapper)`, it introduces a compatibility issue.
If another addon does the same thing, only one of them will win and effectively replace `SaleOrderImportMapper`.

As there is extensibility between backend versions (inheritance through parent backends), it is hard to have modularity for one backend version.

Usually when I encounter a similar issue, my process is the following:
1. Can this feature be integrated in the core module? If yes, it simplifies.
2. If not, I introduce a sort of hook in the core, example: https://github.com/OCA/connector-magento/blob/8.0/magentoerpconnect/sale.py#L430-L435, https://github.com/OCA/connector-magento/blob/8.0/magentoerpconnect/sale.py#L725-L730, in your case it could be SalePricelistImportMapper and SalePricelistImporter. Then I replace only the new ConnectorUnit, reducing the risks of conflicts (still possible but anyway if 2 addons want to change the behavior of the import of pricelists, they are maybe not compatible by design).